### ISSUE TITLE
Unexpected column_names processing when incorporated with line skip

### ISSUE THREAD

**Issue by boothm** [hadley/readxl#178]
I would have expected when skipping lines before starting to read a sheet when column_names is set TRUE, the first line read would be the assumed column_names.  If it is always expected that column_names appear in first row, then it would be good feature add have mechanism to specify the starting row for read rather than having it always assumed equal first row when case is that the header is not in first row.

**Comment by jennybc** [comment]
The behaviour of `skip` is about to align much more with expectations, when #240 gets merged.

Please open a new issue with a reprex if you have continuing difficulty after that.
